File: C++/Valid-Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
    	if (s.size() != t.size()) return false;
        vector<int> vis(26, 0);
        for (int i = 0; i < s.size(); i ++) {
        	vis[s[i] - 'a'] ++;
        }
        for (int i = 0; i < t.size(); i ++) {
        	if (-- vis[t[i]- 'a'] < 0) return false;
        }
        for (int i = 0; i < 26; i ++) {
        	if (vis[i] != 0) return false;
        }
        return true;
    }
};

File: Java/Valid-Anagram.java
public class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        int[] vis = new int[26];
        for (int i = 0; i < s.length(); i ++) {
            vis[s.charAt(i) - 'a'] ++;
        }
        for (int i = 0; i < t.length(); i ++) {
            if (-- vis[t.charAt(i) - 'a'] < 0) return false;
        }
        for (int i = 0; i < 26; i ++) {
            if (vis[i] != 0) return false;
        }
        return true;
    }
}